src/traces/bar/style.js

```diff
@@ -57,7 +95,9 @@ function stylePoints(sel, trace, gd) {
 function styleTextPoints(sel, trace, gd) {
     sel.selectAll('text').each(function(d) {
         var tx = d3.select(this);
-        var font = determineFont(tx, d, trace, gd);
+        var font = Lib.extendFlat({}, determineFont(tx, d, trace, gd), {});
```

Comment:
This looks like a typo. 

```js
var font = determineFont(tx, d, trace, gd);
```

should be good enough, right? If not then, we should move the `extendFlat` call to `determineFont`.

Reply:
A copy is needed since the font.size may be replaced by `minsize`.

Reply:
Ok that explains the

```js
 var font = Lib.extendFlat({}, determineFont(tx, d, trace, gd));
```

part, but isn't third argument in 

```js
var font = Lib.extendFlat({}, determineFont(tx, d, trace, gd), {});
```

useless?

Reply:
Correct. We could/should drop that.
I can make a centralize function for those font getters.
Should `uniform text.minsize` override the font.size(s) defined by traces? If not we could use the minimum among them.

Reply:
> Should uniform text.minsize override the font.size(s) defined by traces?

Like override them in `gd.data[i]` and `gd._fullData[i]`? If so, then the answer is NO. We shouldn't mutate that. 



> If not we could use the minimum among them.

That sounds about right. Adding something like

```js
Lib.applyUniformtext = function(gd, baseFont) {
  var out = Lib.extendFlat({}, baseFont);
  out.size = Math.max(baseFont.size, gd._fullLayout.uniformtext.minsize || 0);
  return out;
}
```

could be a big win!

Reply:
Thanks for the hint.
Would you please clarify which size should be used in the scenario below?
When one of the traces defined its font size to be e.g. 8; while the uniform.minsize is set to 16.
In the current implementation 16 wins over 8.

Reply:
> When one of the traces defined its font size to be e.g. 8; while the uniform.minsize is set to 16.

That's a hard one. I can see arguments for both ways. On one hard, it would make things more flexible if we allow users to override the uniform-text-size results. But then again, if they really want that, they can set all the `textfont.size` in the traces manually w/o even touching `uniformtext`.